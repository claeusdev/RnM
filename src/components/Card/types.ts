import { Character } from "types";

export interface CharacterProps {
    character: Character;
  }