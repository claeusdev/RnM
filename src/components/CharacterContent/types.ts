import { Character, Location } from "types";


export interface CharacterContentProps {
  gender: string;
  location: Location;
  origin: { name: string };
  species: string;
}
