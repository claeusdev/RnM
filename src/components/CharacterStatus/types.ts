export interface CharacterStatusProps {
  status: string;
}
