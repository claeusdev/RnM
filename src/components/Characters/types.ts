import { Character } from "types";

export type CharactersProps = {
    page: string
    results: Character[]
}