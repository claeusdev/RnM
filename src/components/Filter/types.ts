export interface FilterProps {
    updatePageNumber: (pageNumber: number) => void;
}
