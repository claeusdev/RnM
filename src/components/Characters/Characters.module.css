:root {
  --radius: 10px;
  ---light-grey: #fff;
}

.card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  border-radius: 4px;
  box-shadow: rgb(30 41 59 / 4%) 0 2px 4px 0;
  border: 1px solid rgba(98,105,118,.16);
  background: var(---light-grey);
  transition: transform .3s ease-out,opacity .3s ease-out,box-shadow .3s ease-out;
}

.content {
  padding: 10px;
}

.contentHeading {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.characterName {
  font-size: large;
}

.characterSpecies {
  margin-bottom: 2px;
}

.badge {
  top: 10px;
  right: 20px;
  font-size: 17px;
}
