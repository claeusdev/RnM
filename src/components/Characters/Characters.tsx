import CharacterStatus from "components/CharacterStatus";
import { Link } from "react-router-dom";
import styles from "./Characters.module.css";
import { CharactersProps } from "./types";

const Characters = ({ page, results }: CharactersProps) => {
  let content = results.map(({ id, image, name, status, location, species }) => (
    <Link
      style={{ textDecoration: "none" }}
      to={`${page}${id}`}
      key={id}
      className="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
    >
      <div
        className={`${styles.card} d-flex flex-column justify-content-center`}
      >
        <img className={`${styles.img} img-fluid`} src={image} alt="" />
        <div className={styles.content}>
          <div className={styles.contentHeading}>
            <h4 className={styles.characterName}>{name}</h4>
            <CharacterStatus status={status} />
          </div>
          <p className={styles.characterSpecies}>{species}</p>
          
          <div className="">
            <div className="text-small">Last known location:</div>
            <div className="lead">{location.name}</div>
          </div>
        </div>
      </div>

      
    </Link>
  ));
  return <>{content}</>;
};

export default Characters;
