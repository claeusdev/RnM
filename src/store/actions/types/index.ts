export enum ActionTypes {
    LOAD_CHARACTERS = "LOAD_CHARACTERS",
    LOAD_CHARACTERS_SUCCESS = "LOAD_CHARACTERS_SUCCESS",
    LOAD_CHARACTERS_ERROR = "LOAD_CHARACTERS_ERROR",
    FILTER_CHARACTERS = "FILTER_CHARACTERS",
    FILTER_CHARACTERS_SUCCESS = "FILTER_CHARACTERS_SUCCESS",
    FILTER_CHARACTERS_ERROR = "FILTER_CHARACTERS_ERROR",
    LOAD_CHARACTER = "LOAD_CHARACTER",
    LOAD_CHARACTER_SUCCESS = "LOAD_CHARACTER_SUCCESS",
    LOAD_CHARACTER_ERROR = "LOAD_CHARACTER_ERROR",
    SET_EPISODES = "SET_EPISODES"
}