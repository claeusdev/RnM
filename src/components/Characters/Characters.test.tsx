import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { testCharacter } from "types/testData";
import Characters from "./Characters";

test("renders lists of characters", () => {
  const component = render(<BrowserRouter>
    <Characters page="/character/" results={[testCharacter]}/>
  </BrowserRouter>);
  expect(component).toMatchSnapshot();
});
