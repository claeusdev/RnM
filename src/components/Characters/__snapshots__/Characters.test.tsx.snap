// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
        href="/character/1"
        style="text-decoration: none;"
      >
        <div
          class="card d-flex flex-column justify-content-center"
        >
          <img
            alt=""
            class="img img-fluid"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <div
            class="content"
          >
            <div
              class="contentHeading"
            >
              <h4
                class="characterName"
              >
                Rick Sanchez
              </h4>
              <div
                class="text-success"
              >
                Alive
              </div>
            </div>
            <p
              class="characterSpecies"
            >
              Human
            </p>
            <div
              class=""
            >
              <div
                class="text-small"
              >
                Last known location:
              </div>
              <div
                class="lead"
              >
                Citadel of Ricks
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
      href="/character/1"
      style="text-decoration: none;"
    >
      <div
        class="card d-flex flex-column justify-content-center"
      >
        <img
          alt=""
          class="img img-fluid"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <div
          class="content"
        >
          <div
            class="contentHeading"
          >
            <h4
              class="characterName"
            >
              Rick Sanchez
            </h4>
            <div
              class="text-success"
            >
              Alive
            </div>
          </div>
          <p
            class="characterSpecies"
          >
            Human
          </p>
          <div
            class=""
          >
            <div
              class="text-small"
            >
              Last known location:
            </div>
            <div
              class="lead"
            >
              Citadel of Ricks
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders lists of characters 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
        href="/character/1"
        style="text-decoration: none;"
      >
        <div
          class="card d-flex flex-column justify-content-center"
        >
          <img
            alt=""
            class="img img-fluid"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <div
            class="content"
          >
            <div
              class="contentHeading"
            >
              <h4
                class="characterName"
              >
                Rick Sanchez
              </h4>
              <div
                class="text-success"
              >
                Alive
              </div>
            </div>
            <p
              class="characterSpecies"
            >
              Human
            </p>
            <div
              class=""
            >
              <div
                class="text-small"
              >
                Last known location:
              </div>
              <div
                class="lead"
              >
                Citadel of Ricks
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
      href="/character/1"
      style="text-decoration: none;"
    >
      <div
        class="card d-flex flex-column justify-content-center"
      >
        <img
          alt=""
          class="img img-fluid"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <div
          class="content"
        >
          <div
            class="contentHeading"
          >
            <h4
              class="characterName"
            >
              Rick Sanchez
            </h4>
            <div
              class="text-success"
            >
              Alive
            </div>
          </div>
          <p
            class="characterSpecies"
          >
            Human
          </p>
          <div
            class=""
          >
            <div
              class="text-small"
            >
              Last known location:
            </div>
            <div
              class="lead"
            >
              Citadel of Ricks
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
